pkg/asset/ignition/bootstrap/registries.go

```diff
@@ -0,0 +1,33 @@
+package bootstrap
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+func mergedMirrorSets(sources []types.ImageContentSource) []types.ImageContentSource {
+	sourceSet := make(map[string][]string)
+	mirrorSet := make(map[string]sets.String)
+	orderedSources := []string{}
+
+	for _, group := range sources {
+		if _, ok := sourceSet[group.Source]; !ok {
+			orderedSources = append(orderedSources, group.Source)
+			sourceSet[group.Source] = nil
```

Comment:
nit: you don't need to set nil here (the if statement just confirmed it was nil)

Reply:
@patrickdillon we need to set some value in the `map`. the if confirms no key exists for that source.

Reply:
@abhinavdahiya  you can append to the key even if there is no value assigned: 

> Another instance of helpful zero values is a map of slices. Appending to a nil slice just allocates a new slice, so it's a one-liner to append a value to a map of slices; 
([source](https://blog.golang.org/go-maps-in-action))

Perhaps I am missing something else. Unfortunately we can't do that with the set type.